src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.cs

```diff
@@ -231,7 +231,15 @@ private void InitializeOpenBuffers(SnapshotSpan triggerSpan)
                     }
                     Contract.ThrowIfNull(textSnapshot.TextBuffer);
 
-                    openBuffers.Add(textSnapshot.TextBuffer);
+                    // TS creates generated documents to back script blocks in razor generated files.
+                    // These files are opened in the roslyn workspace but are not valid to rename
+                    // as they do not have an undo history.  So we do not include any document that
+                    // we cannot find an undo history for.
+                    var textUndoHistoryService = _workspace.Services.GetService<ITextUndoHistoryWorkspaceService>();
```

Comment:
Should this be moved into TryPopulateOpenTextBufferManagerForBuffer, because this isn't the only place it's called?